### PR TITLE
Windows: Fix tests failing on msvc-cl fastbuild mode

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Commit Message: Windows: Fix tests failing on msvc-cl fastbuild mode
Additional Description:
Fix hystrix_test: SEH exception in mock stub, moved buffer to test class scope to prevent
this

Fix group_verifier_test: absl container assertion tripped due to
lifetime issue for callbacks map, moved to test class scope to prevent
this again

Risk Level: Low
Testing: Modifies unit test setup
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @envoyproxy/windows-dev

**Comment by sunjayBhatia** [comment]
/retest

should be unrelated failures to the change

**Review by wrowe — DISMISSED** [review]
These googletest object lifespans are more correct, should no longer be invalidated on any compiler / optimization level. Patch looks good to me.

**Comment by wrowe** [comment]
/wait

Adding a couple more straightforward fixes before final review

**Comment by sunjayBhatia** [comment]
same old macos failure :(

**Review by wrowe — APPROVED** [review]
This changeset looks complete. The flake still needs to be dealt with in yet another PR/pass through the sources, but seems troublesome since it only occurs in gcp. Good to go on this part.

File: test/extensions/quic_listeners/quiche/integration/BUILD [hunk 1]
```diff
@@ -13,9 +13,9 @@ envoy_cc_test(
     size = "medium",
     srcs = ["quic_http_integration_test.cc"],
     data = ["//test/config/integration/certs"],
-    # TODO(envoyproxy/windows-dev): diagnose msvc-cl build test failure
+    # TODO(envoyproxy/windows-dev): Diagnose msvc-cl opt build gcp rbe CI flake (passes locally)
     tags = [
-        "fails_on_windows",
+        "flaky_on_windows",
```
Comment (sunjayBhatia) [inline test/extensions/quic_listeners/quiche/integration/BUILD:18]: kicked off some runs in RBE, this test target is timing out consistently, though it actually does pass, there is one individual test in particular that takes > 30s `QuicHttpIntegrationTests/QuicHttpIntegrationTest.MultipleQuicConnectionsNoBPF/IPv6_UseGQuicWithTLS`

without the test however this target still takes ~230s

```
//test/extensions/quic_listeners/quiche/integration:quic_http_integration_test PASSED in 227.8s
```

Comment (sunjayBhatia) [reply test/extensions/quic_listeners/quiche/integration/BUILD:18]: On linux this test taks ~30s to fully run:

```
//test/extensions/quic_listeners/quiche/integration:quic_http_integration_test PASSED in 34.6s
```

Comment (sunjayBhatia) [reply test/extensions/quic_listeners/quiche/integration/BUILD:18]: generally the individual tests are slower on Windows, taking ~1-2s each, but looks like all the different versions of `QuicHttpIntegrationTests/QuicHttpIntegrationTest.MultipleQuicConnectionsWithBPF/*` take a while, 20 - 30s on Windows

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**